lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1842,16 +1842,19 @@ func (d *zfs) ListVolumes() ([]Volume, error) {
 			continue // Ignore unrecognised volume.
 		}
 
-		// Detect if a volume is block content type using both the defined suffix and the dataset type.
-		isBlock := strings.HasSuffix(volName, zfsBlockVolSuffix) && zfsContentType == "volume"
+		// Detect if a volume is block content type using only the dataset type.
+		isBlock := zfsContentType == "volume"
```

Comment:
@monstermunchkin did custom block volumes not use the .block suffix ever? or did that change recently?

Reply:
They never have. That's only for VMs as they have two datasets/volumes.

Reply:
> Does this function support detecting filesystem volumes with zfs.block_mode=true?

No, this function only takes the output of `zfs list`. `zfs.block_mode` is a LXD config key and is therefore unknown by this function. There's also no need for this config key as it wouldn't change anything.

Reply:
What this function does not support is listing custom ISO volumes. That is because they follow the same naming schema as custom block volumes. We might want to change this in the future, by suffixing ISOs with `.iso` or something.

Reply:
> No, this function only takes the output of `zfs list`. `zfs.block_mode` is a LXD config key and is therefore unknown by this function. There's also no need for this config key as it wouldn't change anything.

But those filesystem block volumes would be unusable after recovery though right?

Reply:
> What this function does not support is listing custom ISO volumes. That is because they follow the same naming schema as custom block volumes. We might want to change this in the future, by suffixing ISOs with `.iso` or something.

hrm yeah can you open an issue and assign that to you to sort out please.

Reply:
> But those filesystem block volumes would be unusable after recovery though right?

Oh, I see what you mean, and you're right. As a workaround, we could do `lxc storage volume set zfs vol1 zfs.block_mode=true`, right?

Reply:
I dont think you can do that on a volume once it created - if you can thats a bug.

Reply:
We probably need to have a content type suffix for those volumes too